ui/src/app/services/backend-health/backend-health.service.ts

```diff
@@ -0,0 +1,77 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { catchError, from, interval, map, Observable, of, Subject, Subscription, takeUntil } from 'rxjs';
+import { InfodialogComponent } from 'src/app/components/infodialog/infodialog.component';
+import { FetchService } from '../fetch/fetch.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BackendHealthService {
+  private healthCheckSubscription: Subscription = new Subscription;
+  private isBackendHealthy: boolean = true;
+  private unHealthyCheckCount: number = 0;
+
+  constructor(private fetch: FetchService,
+    private dialog: MatDialog) { }
+
+    startHealthCheck() {
+      this.healthCheckSubscription = interval(5000).subscribe(() => {
+        this.checkBackendHealth();
+      });
+    }
+  
+    stopHealthCheck() {
+      if (this.healthCheckSubscription) {
+        this.healthCheckSubscription.unsubscribe();
+      }
+    }
+  
+    checkBackendHealth() {
+      this.checkHealth().subscribe(
+        (isHealthy) => {
+          if (!isHealthy) {
+            if (this.unHealthyCheckCount ==1) {
+              // Backend is unhealthy, open the dialog and unsubscribe
+              this.openHealthDialog();
+            }
+            this.unHealthyCheckCount=1;
+          }
```

Comment:
I think we should have successive health checks failing for a period of X seconds/minutes before showing this to account for possible flakiness. 

Suggest something like 10 consecutive health checks failed so the popup shows up in 50 seconds, or something similar.

Reply:
added 5 successive checks let me know if you think 10 is ideal number